### ISSUE TITLE
Understanding the vod module - Where to start with minio S3

### ISSUE THREAD

**Issue by venomone** [kaltura/nginx-vod-module#1294]
Hey folks,

I really would like to implement this module but I don't really understand where to start as the docs are hard to follow. I want to serve all my HLS segments using minio only, without cloudfront or other AWS services in place! I Only have my application server (WSGI running Django/Python), NGINX/Openresty and minio.

In the end I just want to pass the .m3u8 file to the player and start streaming from a S3 remote location.
I already got in touch with https://github.com/kaltura/nginx-aws-auth-module

Some stuff I also don't understand here, e.g. how does the flow from serving the m3u8 file to the s3 singing of segments work ...
I never worked with HLS before. The only thing I know that I have to return the client a m3u8 playlist and start serving content "somehow" from a s3 remote location.

But still I don't understand how to setup this whole construct. 

Can maybe somebody show his config and tell a little bit about it. Would be awesome :-)

**Comment by erankor** [comment]
There is a sample remote config here -
https://github.com/kaltura/nginx-vod-module#remote-configuration

You can take this config, and change the location that does the `proxy_pass` to look like the sample location here -
https://github.com/kaltura/nginx-aws-auth-module#sample-configuration
(in order to generate the authentication headers)

I'd suggest to start with that, and use some trial & error... the messages in the nginx error log are usually easy to understand.

**Comment by venomone** [comment]
@erankor Thanks for your response. I'm using minio to serve HLS segments. Is there any way i can secure the segment and the location? Basically the usage of secure_link module would be awesome. any idea?
Beside, can you please explain to my what ~ ^/p/\d+/(sp/\d+/)?serveFlavor/ actually does?!


    # vod settings
    vod_mode remote;
    vod_upstream_location /streaming;
    vod_last_modified 'Sun, 19 Nov 2000 08:52:00 GMT';
    vod_last_modified_types *;

    # vod caches
    vod_metadata_cache metadata_cache 512m;
    vod_response_cache response_cache 128m;

    # gzip manifests
    gzip on;
    gzip_types application/vnd.apple.mpegurl;

    location /streaming {
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_pass http://minio0:9000/Movies;
    }

    location ~ ^/p/\d+/(sp/\d+/)?serveFlavor/ {
            vod hls;

            add_header Access-Control-Allow-Headers '*';
            add_header Access-Control-Expose-Headers 'Server,range,Content-Length,Content-Range';
            add_header Access-Control-Allow-Methods 'GET, HEAD, OPTIONS';
            add_header Access-Control-Allow-Origin '*';
            expires 100d;
    }
